### ISSUE TITLE
Commands with quotes fail to parse

### ISSUE THREAD

**Issue by michalrus** [schell/steeloverseer#30]
On 85d40083ac893bebc3696ab48f223da8af928874, e.g.:

```
$ echo "cabal test --test-options='1234'\\'' 567'"
cabal test --test-options='1234'\'' 567'

$ sos . -c "cabal test --test-options='1234'\\'' 567'"
sos: Error parsing command 'cabal test --test-options='1234'\'' 567''

(1)$ 
```

**Comment by schell** [comment]
@michalrus what's the use case here?

**Comment by michalrus** [comment]
@schell, say, if I wanted to match a Tasty test case with a space in name, I’d need that. The pure Cabal command would then be:

```
cabal test --test-options='-p '\''some test name w/ spaces'\'' --color=always'
```

Currently there’s no way to pass that command to `sos -c [ … here … ]`.

**Comment by schell** [comment]
Yes, I see. The difficulty here is that within a command we use `\` as the leader to a capture group. It's possible we simply need some logic around that to check for quotes, as that is obviously not a valid capture group.

I'll get this when I can but until then I welcome any pull request, @michalrus! Here's where the logic should probably go:
https://github.com/schell/steeloverseer/blob/master/src/Sos/Template.hs#L41
